flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveShimV110.java

```diff
@@ -19,279 +19,55 @@
-import org.apache.hadoop.hive.serde2.objectinspector.primitive.WritableConstantDateObjectInspector;
-import org.apache.hadoop.hive.serde2.objectinspector.primitive.WritableConstantDoubleObjectInspector;
-import org.apache.hadoop.hive.serde2.objectinspector.primitive.WritableConstantFloatObjectInspector;
-import org.apache.hadoop.hive.serde2.objectinspector.primitive.WritableConstantHiveCharObjectInspector;
-import org.apache.hadoop.hive.serde2.objectinspector.primitive.WritableConstantHiveDecimalObjectInspector;
-import org.apache.hadoop.hive.serde2.objectinspector.primitive.WritableConstantHiveVarcharObjectInspector;
-import org.apache.hadoop.hive.serde2.objectinspector.primitive.WritableConstantIntObjectInspector;
-import org.apache.hadoop.hive.serde2.objectinspector.primitive.WritableConstantLongObjectInspector;
-import org.apache.hadoop.hive.serde2.objectinspector.primitive.WritableConstantShortObjectInspector;
-import org.apache.hadoop.hive.serde2.objectinspector.primitive.WritableConstantStringObjectInspector;
-import org.apache.hadoop.hive.serde2.objectinspector.primitive.WritableConstantTimestampObjectInspector;
-import org.apache.hadoop.hive.serde2.typeinfo.CharTypeInfo;
-import org.apache.hadoop.hive.serde2.typeinfo.PrimitiveTypeInfo;
-import org.apache.hadoop.hive.serde2.typeinfo.VarcharTypeInfo;
-import org.apache.thrift.TException;
+import org.apache.hadoop.hive.ql.exec.FileSinkOperator;
+import org.apache.hadoop.hive.serde2.Deserializer;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.hadoop.mapred.OutputFormat;
+import org.apache.hadoop.mapred.Reporter;
 
-import java.io.IOException;
-import java.lang.reflect.Constructor;
-import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
-import java.util.ArrayList;
 import java.util.List;
-import java.util.Map;
+import java.util.Properties;
 
 /**
  * Shim for Hive version 1.1.0.
  */
-public class HiveShimV110 implements HiveShim {
+public class HiveShimV110 extends HiveShimV101 {
 
 	@Override
-	public IMetaStoreClient getHiveMetastoreClient(HiveConf hiveConf) {
+	public FileSinkOperator.RecordWriter getHiveRecordWriter(JobConf jobConf, String outputFormatClzName,
+			Class<? extends Writable> outValClz, boolean isCompressed, Properties tableProps, Path outPath) {
 		try {
-			return new HiveMetaStoreClient(hiveConf);
-		} catch (MetaException ex) {
-			throw new CatalogException("Failed to create Hive Metastore client", ex);
+			Class outputFormatClz = Class.forName(outputFormatClzName);
+			Class utilClass = Class.forName("org.apache.hadoop.hive.ql.io.HiveFileFormatUtils");
+			Method utilMethod = utilClass.getDeclaredMethod("getOutputFormatSubstitute", Class.class);
+			outputFormatClz = (Class) utilMethod.invoke(null, outputFormatClz);
+			OutputFormat outputFormat = (OutputFormat) outputFormatClz.newInstance();
```

Comment:
Nit: add a similar assert as in 1.0.0

Reply:
Newer version can't return null judging by the code. But I guess it's no harm to assert it anyway.